src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/format/DateFieldFormatter.java

```diff
@@ -0,0 +1,96 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.executor.format;
+
+import org.apache.commons.lang3.time.DateUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.text.ParseException;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+public class DateFieldFormatter {
+    private static final Logger LOG = LogManager.getLogger(DateFieldFormatter.class);
+    private static final String FORMAT_JDBC = "yyyy-MM-dd HH:mm:ss.SSS";
+
+    private static final String FORMAT_DOT_DATE_AND_TIME = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+    private static final String FORMAT_DOT_KIBANA_SAMPLE_DATA_LOGS_EXCEPTION = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
+    private static final String FORMAT_DOT_KIBANA_SAMPLE_DATA_FLIGHTS_EXCEPTION = "yyyy-MM-dd'T'HH:mm:ss";
+    private static final String FORMAT_DOT_KIBANA_SAMPLE_DATA_ECOMMERCE_EXCEPTION = "yyyy-MM-dd'T'HH:mm:ssXXX";
+    private static final String FORMAT_DOT_DATE = "yyyy-MM-dd";
+
+    private final Map<String, String> dateFieldFormatMap;
+    private List<Schema.Column> columns;
+
+    public DateFieldFormatter(Map<String, String> dateFieldFormatMap, List<Schema.Column> columns) {
+        this.dateFieldFormatMap = dateFieldFormatMap;
+        this.columns = columns;
+    }
+
+    public void applyJDBCDateFormat(Map<String, Object> rowSource) {
+        for (Schema.Column column : columns) {
```

Comment:
I think you should pick columns of `DATE` type into a `Set` in your constructor to save this for loop. Otherwise it runs for each row in the result set.

Reply:
Now saving date columns in constructor (saving as `List`, since it seems that `Set`s cannot be iterated through with this foreach syntax

Reply:
Thanks for the changes! But I assume a `Set<String>` with all field names of `DATE` type is sufficient right? Why original `Column` saved in a list with for-loop ran for each row in result set?

Reply:
Oh yes, this makes sense! Since all I need is the name of the column. Will make this change

Reply:
Addressed (now only storing a `Set<String>` for date column names)